lib/matplotlib/image.py

```diff
@@ -56,11 +56,131 @@
     'blackman': _image.BLACKMAN,
 }
 
-
 interpolations_names = set(six.iterkeys(_interpd_))
 
 
-class _AxesImageBase(martist.Artist, cm.ScalarMappable):
+def composite_images(images, renderer, magnification=1.0):
+    """
+    Composite a number of RGBA images into one.  The images are
+    composited in the order in which they appear in the `images` list.
+
+    Parameters
+    ----------
+    images : list of Images
+        Each must have a `make_image` method.  For each image,
+        `can_composite` should return `True`, though this is not
+        enforced by this function.  Each image must have a purely
+        affine transformation with no shear.
+
+    renderer : RendererBase instance
+
+    magnification : float
+        The additional magnification to apply for the renderer in use.
+
+    Returns
+    -------
+    tuple : image, offset_x, offset_y
+        Returns the tuple:
+
+        - image: A numpy array of the same type as the input images.
+
+        - offset_x, offset_y: The offset of the image (left, bottom)
+          in the output figure.
+    """
+    if len(images) == 0:
+        return np.empty((0, 0, 4), dtype=np.uint8), 0, 0
+
+    parts = []
+    bboxes = []
+    for image in images:
+        data, x, y, trans = image.make_image(renderer, magnification)
+        if data is not None:
+            x *= magnification
+            y *= magnification
+            parts.append((data, x, y, image.get_alpha() or 1.0))
+            bboxes.append(
+                Bbox([[x, y], [x + data.shape[1], y + data.shape[0]]]))
+
+    if len(parts) == 0:
+        return np.empty((0, 0, 4), dtype=np.uint8), 0, 0
+
+    bbox = Bbox.union(bboxes)
+
+    output = np.zeros(
+        (int(bbox.height), int(bbox.width), 4), dtype=np.uint8)
+
+    for data, x, y, alpha in parts:
+        trans = Affine2D().translate(x - bbox.x0, y - bbox.y0)
+        _image.resample(data, output, trans, _image.NEAREST,
+                        resample=False, alpha=alpha)
+
+    return output, bbox.x0 / magnification, bbox.y0 / magnification
+
+
+def _draw_list_compositing_images(
+        renderer, parent, dsu, suppress_composite=None):
+    """
+    Draw a sorted list of artists, compositing images into a single
+    image where possible.
+
+    For internal matplotlib use only: It is here to reduce duplication
+    between `Figure.draw` and `Axes.draw`, but otherwise should not be
+    generally useful.
+    """
+    has_images = np.any(isinstance(x[1], _ImageBase) for x in dsu)
```

Comment:
Thanks, guys.  Good catch.  I think we get something that evaluates always as True...